### PR TITLE
noun query timeout

### PR DESCRIPTION
**Pull request by daveads**
Fix for this https://github.com/scribe-org/Scribe-Data/issues/124

### CODE REVIEW

File: src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql [hunk 1]
```diff
@@ -1,45 +1,40 @@
 # tool: scribe-data
-# All German (Q188) nouns, their plural and their gender.
+# All German (Q188) nouns, their plural, and their gender.
```
Comment (andrewtavis) [inline src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql:2]: No need to change this as we don't use Oxford commas :)

File: src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql [hunk 2]
```diff
@@ -1,45 +1,40 @@
...
-  } .
-
-  # Optional selection of nominative plural forms.
+  # Optional selection of nominative plural form.
   OPTIONAL {
     ?lexeme ontolex:lexicalForm ?pluralForm .
     ?pluralForm ontolex:representation ?plural ;
-      wikibase:grammaticalFeature wd:Q146786 ;
-      wikibase:grammaticalFeature wd:Q131105 ;
+                wikibase:grammaticalFeature wd:Q146786 ;
```
Comment (andrewtavis) [inline src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql:23]: No need to reformat.

File: src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql [hunk 3]
```diff
@@ -1,45 +1,40 @@
...
+                wikibase:grammaticalFeature wd:Q131105 ;
   } .
 
-  # Optional selection of genders.
+  # Optional selection of gender.
```
Comment (andrewtavis) [inline src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql:27]: Let's leave it as genders as there could be words that have more than one :)

File: src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql [hunk 4]
```diff
@@ -1,45 +1,40 @@
...
   } .
 
+  # Get the label of the gender.
   SERVICE wikibase:label {
     bd:serviceParam wikibase:language "[AUTO_LANGUAGE]".
     ?nounGender rdfs:label ?gender .
   }
 }
+LIMIT 10000
+OFFSET OFFSET_BY
```
Comment (andrewtavis) [inline src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql:40]: I likely will add in an `OFFSET_AMOUNT` here and replace it, but no stress. I'll get to it 😊

File: src/scribe_data/extract_transform/wikidata/update_data.py [hunk 5]
```diff
@@ -130,108 +136,46 @@
...
+    
+    
+    if any("OFFSET" in line for line in query_lines):
+        
+        language_id = extract_language_id(query_lines)
+        Total_nouns = get_total_nouns(language_id)
+
+        batch_size = 10000  #adjust as needed {{Same as limit}}
+     
+        # Calculate the number of iterations needed
```
Comment (andrewtavis) [inline src/scribe_data/extract_transform/wikidata/update_data.py:148]: Minor nitpick on this, if the comment is it's own line, then it's Python standard to include a period :) I can add these, but for later 🚀

File: src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql (hunk 3 above)
Comment (daveads) [reply src/scribe_data/extract_transform/languages/German/nouns/query_nouns.sparql:27]: alright

File: src/scribe_data/extract_transform/wikidata/update_data.py (hunk 5 above)
Comment (daveads) [reply src/scribe_data/extract_transform/wikidata/update_data.py:148]: alright

**Comment by daveads** [comment]
hmm...

**Comment by daveads** [comment]
@andrewtavis is there any probs with this pr ?

**Comment by andrewtavis** [comment]
I haven't been able to give it a full check yet, @daveads, but I'll do it later this evening and fix anything that needs to be fixed :) Thanks for checking in!

**Comment by daveads** [comment]
> I haven't been able to give it a full check yet, @daveads, but I'll do it later this evening and fix anything that needs to be fixed :) Thanks for checking in!

okay

**Review by andrewtavis — APPROVED** [review]
I ended up realizing that it doesn't make sense to figure out update_data.py without also doing the directory restructure. This does the restructure, and I added back in parts of the code that was sent along.

Sorry that this is a bit of a mess, @daveads. Just a lot going on. I'll go ahread and fix the rest of this post merge 😊

**Pull request merged**